crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -1067,10 +1099,12 @@ impl<DB: Database, EF: ExecutorFactory> BlockchainTree<DB, EF> {
         }
 
         let head = chain_notification.tip().header.clone();
-
+        let sealed_block = chain_notification.tip().block.clone();
         // send notification about new canonical chain.
         let _ = self.canon_state_notification_sender.send(chain_notification);
 
+        self.pending_block.try_update(Some(sealed_block));
+
```

Comment:
this is not quite right because here we're setting the pending block to the tip

we need to set it None, or check if we have a pending block in the tree already